src/xml-properties-mixin.js

```diff
@@ -0,0 +1,151 @@
+const re = require('block-re');
+const cheerio = require('cheerio');
+
+const makeNSTagSelector = require('./make-ns-tag-selector');
+
+const $ = cheerio.load(`<?xml version="1.0" ?>`, { xmlMode: true });
+
+//
+// mixin to persist data in an xml properties element, similar to how properties
+// are stored in pPr/rPr elements in ooxml.
+//
+// numbers and strings are stored as the 'val' attribute of an element, e.g.
+// properties.foo = 5 ==> <foo val="5"/>
+//
+// key: value objects
+// properties.foo = { bar: 5, baz: 'bat' } ==> <foo bar="5", baz="bat"/>
+//
+// array elements
+// properties.push('foo', {bar: 5})
+// properties.push('foo', {bar: 6})
+// ==> <foo bar="5"/><foo bar="6"/>
+//
+
+let XmlPropertiesMixin = (superclass) => class extends superclass {
```

Comment:
Can you add brief comments to each of these methods (even the internal helpers) explaining what they do? Right now I kind have to read the whole file to figure our what one particular method is doing.